components/ipfs/ipfs_file_import_worker.cc

```diff
@@ -0,0 +1,98 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/ipfs/ipfs_file_import_worker.h"
+
+#include <utility>
+
+#include "base/files/file_util.h"
+#include "base/guid.h"
+#include "base/task/post_task.h"
+#include "base/task_runner_util.h"
+#include "brave/components/ipfs/import_utils.h"
+#include "brave/components/ipfs/ipfs_constants.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/browser_thread.h"
+#include "net/base/mime_util.h"
+#include "services/network/public/cpp/shared_url_loader_factory.h"
+#include "services/network/public/cpp/simple_url_loader.h"
+#include "storage/browser/blob/blob_data_builder.h"
+#include "url/gurl.h"
+
+namespace {
+
+const char kFileValueName[] = "file";
+
+const char kDefaultMimeType[] = "application/octet-stream";
+
+int64_t CalculateFileSize(base::FilePath upload_file_path) {
```

Comment:
Hm, I meant in my previous review comment for all anonymous functions to be shared if possible, looks like `CalculateFileSize` and `BuildBlobWithFile` can be shared too?
Also, when you move the function into utils, please also double check to remove header includes which might be no longer needed from this file.